torch/distributed/_tensor/placement_types.py

```diff
@@ -72,7 +72,8 @@ def _split_tensor(
 
     def _pad_tensor(self, tensor: torch.Tensor) -> torch.Tensor:
         # pad tensor by 1 on the shard dim
-        pad = [0, 0] * (tensor.ndim - self.dim)
+        pad_num = -self.dim if self.dim < 0 else tensor.ndim - self.dim
```

Comment:
So this is dealing with negative dims? My feeling is that we might want to keep Shard placement `dim` be always positive (unless specific reason) in our internal op implementation, if it's negative from user, we should call `normalize_dim` before constructing `Shard` placement, this is so that we deal with `negative_dim` in a uniform manner, see https://github.com/pytorch/pytorch/blob/60e503d4685f7cb4e7b7af5b11548c1d47d299e0/torch/distributed/_tensor/ops/utils.py#L62